src/Data/Aeson/TextMap.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE TemplateHaskell #-}
+module Data.Aeson.TextMap (
+    TextMap,
+    difference,
+    empty,
+    foldrWithKey,
+    fromHashMap,
+    fromList,
+    fromListWith,
+    insert,
+    keys,
+    lookup,
+    mapKeyVal,
+    singleton,
+    size,
+    toList,
+    toAscList,
+    toHashMap,
+    traverseWithKey
+) where
+
+#if 1
+import Control.DeepSeq (NFData(..))
+import Data.Data (Data)
+import Data.Hashable (Hashable(..))
+import Data.HashMap.Strict (HashMap)
+import Data.List (sortBy)
+import Data.Ord (comparing)
+import Data.Text (Text, unpack)
+import Data.Typeable (Typeable)
+import Prelude hiding (lookup)
+import Data.Bifunctor (first)
+
+
+import qualified Data.HashMap.Strict as H
+import qualified Language.Haskell.TH.Syntax as TH
+
+newtype TextMap v = TextMap { unTextMap :: HashMap Text v }
+  deriving (Eq, Ord, Typeable, Data)
+  deriving newtype (Read, Show, Functor, Hashable, NFData)
```

Comment:
`DerivingStrategies` is GHC-8.2 only. We should write these instance by hand for now. (stock deriving for `Functor` is fine. We need `Foldable` and `Traversable` too, latter cannot be `newtype` derivied anyway: https://ryanglscott.github.io/2018/06/22/quantifiedconstraints-and-the-trouble-with-traversable/).